spec/unit/promstats_spec.rb

```diff
@@ -0,0 +1,269 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'prometheus' do
+  logger = MockLogger.new
+  params = { 'prefix': 'test', 'metrics_prefix': 'mtest', 'endpoint': 'eptest' }
+  subject = Vmpooler::Metrics::Promstats.new(logger, params)
+  let(:logger) { MockLogger.new }
+
+  describe '#initialise' do
+    it 'returns a Metrics object' do
+      expect(Vmpooler::Metrics::Promstats.new(logger)).to be_a(Vmpooler::Metrics)
+    end
+  end
+
+  describe '#find_metric' do
+    context "Single Value Parameters" do
+      let!(:foo_metrics) do
+        { metric_suffixes: { bar: 'baz' },
+          param_labels: %i[first second last] }
+      end
+      let!(:labels_hash) { { labels: { :first => nil, :second => nil, :last => nil } } }
+      before { subject.instance_variable_set(:@p_metrics, { foo: foo_metrics }) }
+  
+      it 'returns the metric for a given label including parsed labels' do
+        expect(subject.find_metric('foo.bar')).to include(metric_name: '_bar')
+        expect(subject.find_metric('foo.bar')).to include(foo_metrics)
+        expect(subject.find_metric('foo.bar')).to include(labels_hash)
+        # I would expect this to fail because 'bar' is not a direct subkey of 'foo'.
```

Comment:
Is this comment still relevant? I dont understand the comment.

Reply:
Removed it